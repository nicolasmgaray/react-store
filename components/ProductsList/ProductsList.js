import React from "react";
import { List } from "../../styles/Products";
import ProductItem from "./ProductItem";

const ProductList = ({ products, points }) => (
  <List>
    {products.map(x => (
      <ProductItem key={x._id} {...x} points={points}></ProductItem>
    ))}
  </List>
);

export default ProductList;
