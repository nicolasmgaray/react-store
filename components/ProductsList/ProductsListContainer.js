import React from "react";
import { useSelector } from "react-redux";
import ProductList from "./ProductsList";
import Spinner from "../UI/Spinner";
import filterFunctions from "../../utils/filters";
const {
  orderByHighest,
  orderByLowest,
  orderByRecent,
  filterByTerm
} = filterFunctions;

const ProductListContainer = () => {
  const { products: productsState, user, filters } = useSelector(
    state => state
  );
  const { fetching, products } = productsState;

  let filteredProducts = filterByTerm(products, filters.searchTerm);

  switch (filters.orderBy) {
    case "lowest":
      filteredProducts = orderByLowest(filteredProducts);
      break;
    case "highest":
      filteredProducts = orderByHighest(filteredProducts);
      break;
    default:
      filteredProducts = orderByRecent(filteredProducts);
      break;
  }

  if (fetching) return <Spinner></Spinner>;
  else
    return (
      <ProductList
        points={user.points}
        products={filteredProducts}
      ></ProductList>
    );
};

export default ProductListContainer;
