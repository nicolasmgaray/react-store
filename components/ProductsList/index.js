import ProductsList from "./ProductsListContainer";
export default ProductsList;