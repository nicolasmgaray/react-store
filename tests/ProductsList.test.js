import React from "react";
import { shallow } from "enzyme";
import ProductsList from "../components/ProductsList/ProductsList.js";

describe("ProductsList", () => {
  const props = {
    points: 5000,
    products: [
      {
        _id: "5a0b35c1734d1d08bf7084c3",
        name: "iPhone 8",
        cost: 800,
        category: "Phones",
        img: {
          url: "https://aerolab-challenge.now.sh/images/iPhone8-x1.png",
          hdUrl: "https://aerolab-challenge.now.sh/images/iPhone8-x2.png"
        }
      },
      {
        _id: "5a0b35d7734d1d08bf7084c9",
        name: "Nintendo Switch 32GB",
        cost: 300,
        category: "Gaming",
        img: {
          url: "https://aerolab-challenge.now.sh/images/Switch-x1.png",
          hdUrl: "https://aerolab-challenge.now.sh/images/Switch-x2.png"
        }
      }
    ]
  };

  it("renders without crashing", () => {
    shallow(<ProductsList {...props} />);
  });
  it("matches snapshot", () => {
    const component = shallow(<ProductsList {...props} />);
    expect(component).toMatchSnapshot();
  });
});
