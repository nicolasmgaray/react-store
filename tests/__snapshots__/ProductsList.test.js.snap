// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList matches snapshot 1`] = `ShallowWrapper {}`;
